cmd/cg-client/main.go

```diff
@@ -0,0 +1,342 @@
+		if err != nil {
+			return err
+		}
+		file.Close()
+	}
+
+	credFile, err := ini.Load(credentialFilename)
+	if err != nil {
+		return err
+	}
+
+	// Setup HTTPS client
+	tlsConfig := &tls.Config{
+		Certificates: []tls.Certificate{cert},
+	}
+	tlsConfig.BuildNameToCertificate()
+	transport := &http.Transport{TLSClientConfig: tlsConfig}
+
+	// proxy env variables in ascending order of preference, lower case 'http_proxy' dominates
+	// just like curl
+	proxyEnvVariables := []string{"HTTP_PROXY", "HTTPS_PROXY", "http_proxy"}
+	for _, proxyVar := range proxyEnvVariables {
+		httpProxy, err := getParseURLEnvVariable(proxyVar)
+		if err == nil && httpProxy != nil {
+			transport.Proxy = http.ProxyURL(httpProxy)
+		}
+	}
+
+	client := &http.Client{Transport: transport, Timeout: 10 * time.Second}
+
+	// Do GET something
+	resp, err := client.Get(baseUrl)
+	if err != nil {
+		log.Printf("failed to connect err=%s transport=%+v ", err, transport)
+		if resp != nil {
+			log.Printf("resp=+%v", resp)
+		}
+		log.Fatal(err)
+	}
+	defer resp.Body.Close()
+
+	// Dump response
+	data, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		log.Fatal(err)
+	}
+	//log.Println(string(data))
+
+	var accountList getAccountInfo
+	err = json.Unmarshal(data, &accountList)
+	if err != nil {
+		log.Fatal(err)
+	}
+	log.Printf("%+v", accountList)
+
+	for _, account := range accountList.CloudAccounts {
+		for _, roleName := range account.AvailableRoles {
+			adminRole, err := regexp.Match("(?i)admin", []byte(roleName))
+			if adminRole && !askAdminRoles {
+				continue
+			}
+			if err != nil {
+				log.Fatalf("error on regexp=%s", err)
+			}
+			err = getAndUptateCreds(client, baseUrl,
+				account.Name, roleName, credFile,
+				outputProfilePrefix, lowerCaseProfileName)
+			if err != nil {
+				if err.Error() == badReturnErrText {
+					log.Printf("skipping role")
+					continue
+				}
+				log.Fatalf("error on getAnd UpdateCreds=%s", err)
+			}
+		}
+	}
+	err = credFile.SaveTo(credentialFilename)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	return nil
+}
+
+//Assumes cert is pem ecoded
+func getCertExpirationTime(certFilename string) (time.Time, error) {
+	dat, err := ioutil.ReadFile(certFilename)
+	if err != nil {
+		return time.Now(), err
+	}
+	block, _ := pem.Decode(dat)
+
+	cert, err := x509.ParseCertificate(block.Bytes)
+	if err != nil {
+		return time.Now(), err
+	}
+	return cert.NotAfter, nil
+}
+
+func getUserHomeDir() (homeDir string) {
+	homeDir = os.Getenv("HOME")
+	if homeDir != "" {
+		return homeDir
+	}
+	usr, err := user.Current()
+	if err != nil {
+		return homeDir
+	}
+	// TODO: verify on Windows... see: http://stackoverflow.com/questions/7922270/obtain-users-home-directory
+	homeDir = usr.HomeDir
+	return
+}
+
+func Usage() {
+	fmt.Fprintf(
+		os.Stderr, "Usage of %s (version %s):\n", os.Args[0], Version)
+	flag.PrintDefaults()
+}
+
+func main() {
+	flag.Usage = Usage
+	flag.Parse()
+
+	err := saveDefaultConfig(*configFilename)
```

Comment:
Why unconditionally save? Shouldn't the logic be to try loading and if that fails, maybe use a built-in default (which doesn't have company-specific information in the source code).
Why not inject site-specific configuration information just like the `Version` variable data is being injected? It could be read from a file that listed in `.gitignore` so that it never gets checked into the repository, or instead it could come from an environment variable.